lib/Doctrine/Common/Annotations/AnnotationRegistry.php

```diff
@@ -111,9 +111,11 @@ public static function registerAutoloadNamespaces(array $namespaces) : void
      */
     public static function registerLoader(callable $callable) : void
     {
-        // Reset our static cache now that we have a new loader to work with
-        self::$failedToAutoload   = [];
-        self::$loaders[]          = $callable;
+        if ($callable !== 'class_exists' || !self::isLastLoaderClassExists()) {
```

Comment:
Please invert this condition and make it early-return. Also space around `!` is required.